gobblin-core/src/main/java/gobblin/source/extractor/extract/jdbc/OracleExtractor.java

```diff
@@ -0,0 +1,402 @@
+    String query = this.getExtractSql();
+
+    if (StringUtils.isBlank(watermarkFilter)) {
+      watermarkFilter = "1=1";
+    }
+    query = query.replace(this.getOutputColumnProjection(), columnProjection)
+        .replace(ConfigurationKeys.DEFAULT_SOURCE_QUERYBASED_WATERMARK_PREDICATE_SYMBOL, watermarkFilter);
+    String sampleFilter = this.constructSampleClause();
+
+    if (!StringUtils.isEmpty(sampleFilter)) {
+      String col = sampleFilter + " 1 as col ";
+      query = "SELECT COUNT(1) FROM (" + query.replace(" COUNT(1) ", col) + ")temp";
+    }
+    commands.add(JdbcExtractor.getCommand(query, JdbcCommand.JdbcCommandType.QUERY));
+    return commands;
+  }
+
+  @Override
+  public List<Command> getDataMetadata(String schema, String entity, WorkUnit workUnit, List<Predicate> predicateList)
+      throws DataRecordException {
+    log.debug("Build query to extract data");
+    List<Command> commands = new ArrayList<>();
+    int fetchSize = this.workUnitState.getPropAsInt(ConfigurationKeys.SOURCE_QUERYBASED_JDBC_RESULTSET_FETCH_SIZE,
+        ConfigurationKeys.DEFAULT_SOURCE_QUERYBASED_JDBC_RESULTSET_FETCH_SIZE);
+    log.info("Setting jdbc resultset fetch size as " + fetchSize);
+
+    String watermarkFilter = this.concatPredicates(predicateList);
+    String query = this.getExtractSql();
+    if (StringUtils.isBlank(watermarkFilter)) {
+      watermarkFilter = "1=1";
+    }
+
+    query = query.replace(ConfigurationKeys.DEFAULT_SOURCE_QUERYBASED_WATERMARK_PREDICATE_SYMBOL, watermarkFilter);
+    String sampleFilter = this.constructSampleClause();
+
+    if (!StringUtils.isEmpty(sampleFilter)) {
+      String columnProjection = this.getOutputColumnProjection();
+      String newColumnProjection = sampleFilter + " " + columnProjection;
+      query = query.replace(columnProjection, newColumnProjection);
+    }
+
+    commands.add(JdbcExtractor.getCommand(query, JdbcCommand.JdbcCommandType.QUERY));
+    commands.add(JdbcExtractor.getCommand(fetchSize, JdbcCommand.JdbcCommandType.FETCHSIZE));
+    return commands;
+  }
+
+  // TODO: How to handle variable precision datatypes e.g. NUMBER(6)
+  @Override
+  public Map<String, String> getDataTypeMap() {
+    Map<String, String> dataTypeMap = ImmutableMap.<String, String> builder().put("char", "string")
+        .put("varchar2", "string").put("varchar", "string").put("nchar", "string").put("nvarchar2", "string")
+        .put("nclob", "string").put("clob", "string").put("long", "string").put("number", "double")
+        .put("binary_float", "float").put("binary_double", "double").put("date", "date").put("datetime", "timestamp")
+        .put("timestamp", "timestamp").put("timestamp with time zone", "timestamp").put("numeric", "double")
+        .put("timezone with local timezone", "timestamp").put("raw", "string").put("long raw", "string")
+        .put("rowid", "string").put("urowid", "string").put("xmltype", "string").put("float", "float")
```

Comment:
@bjvanov Could you remove .put("float", "float") entry? It's currently throwing out IllegalArgumentException due to the conflict with new addition. Also, it would be great if you can add link to the doc mentioning this float --> double mapping.